query-node/joystream-query-node/schema.graphql

```diff
@@ -1,21 +1,212 @@
+
+	"All properties that have been used on this class across different class schemas"
+	properties: [Property!]
+
+	"All schemas that are available for this class, think v0.0 Person, v.1.0 Person, etc."
+	schemas: [Schema!] @derivedFrom(field: "class")
+
+	"Current class maintainer curator groups"
+	maintainers: [CuratorGroup!]
+
+	"All vouchers for this class"
+	vouchers: [EntityCreationVoucher!] @derivedFrom(field: "class")
+}
+
+enum PropertyType {
+	BOOL
+	UINT16
+	UINT32
+	UINT64
+	INT16
+	INT32
+	INT64
+	TEXT
+	REFERENCE
+	VECBOOL
+	VECUINT16
+	VECUINT32
+	VECUINT64
+	VECINT16
+	VECINT32
+	VECINT64
+	VECTEXT
+	VECREFERENCE
+}
+
+"`Property` representation, related to a given `Class`"
+type Property @entity {
+	"Runtime identifier for property in class"
+	propertyID: BigInt!
+
+	"Type of `Property`"
+	propertyType: PropertyType!
+
+	"If property value can be skipped, when adding entity schema support"
+	required: Boolean!
+
+	"Used to enforce uniqeness of a property across all entities that have this property"
+	unique: Boolean!
+
+	"Name of property"
+	name: String!
+
+	"Description of property"
+	description: String!
+
+	"Property is locked from maintainer"
+	isLockedFromMaintainer: Boolean!
+
+	"Property is locked from controller"
+	isLockedFromController: Boolean!
+
+	"Upper bound for vector length when property is a vector"
+	maximumLength: BigInt
+
+	"Optional field, it must reference a class then PropertyType is Reference"
+	referencedClass: Class
+
+	"Optional field, Only will be set when PropertyType is Reference"
+	sameControllerStatus: Boolean
+
+	"Schemas which involve this property"
+	schemas: [Schema!] @derivedFrom(field: "properties")
+}
+
+"Represents a standard for what information must be associated with any entity of a given class"
+type Schema @entity {
+	"Runtime identifier of schema in class"
+	schemaId: BigInteger!
+
+	"Class to which this schema corresponds"
+	class: Class!
+
+	"Properties of class that are in this schema"
+	properties: [Property!]
+
+	"If schema can be added to an entity"
+	isActive: Boolean!
+}
+
+"Working group participant: working-group/src/types.rs"
+type Worker @entity {
+	member: Member!
+
+	"Account used to authenticate in this role"
+	roleAccountId: Bytes!
+
+	# "Whether the role has recurring reward, and if so an identifier for this"
+	# rewardRelationship: Option<RewardRelationshipId>
+
+	# "When set, describes role stake of the worker/lead"
+	# roleStakeProfile: Option<RoleStakeProfile<StakeId, BlockNumber>>
+}
+
+type Curator @entity {
+	"Account used to authenticate"
+	worker: Worker!
+
+	groups: [CuratorGroup] @derivedFrom(field: "curators")
+}
+
+"A group, that consists of `curators` set"
+type CuratorGroup @entity {
+	"Runtime identifier for group"
+	groupId: BigInt!
+
+	"Activity stats of group"
+	isActive: Boolean!
+
+	"All member curators of this group"
+	curators: [Curator!]
+
+	"All classes this group maintains"
+	maintainerOfClasses: [Class!] @derivedFrom(field: "maintainers")
```

Comment:
We only should know the number of classes, maintained by this curator group, don`t we?

```
pub struct CuratorGroup<T: Trait> {
    /// Curators set, associated with a iven curator group
    curators: BTreeSet<T::CuratorId>,

    /// When `false`, curator in a given group is forbidden to act
    active: bool,

    /// Used to count the number of `Class`(es), given curator group maintains
    number_of_classes_maintained: u32,
}
```

Reply:
Yes right, but instead of having only a number we create a sql reference to those maintainers. Let's say someone needs to get all maintainers data related to maintainers ie: 
```graphql
query {
  class(classId: 123) {
    maintainers {
      groupId
      isActive
    }
  }
}
```

Reply:
ok, good to know!